modules/EnsEMBL/Web/Component/Gene/ComparaTree.pm

```diff
@@ -280,14 +280,17 @@ sub content {
     my @highlight_tags             = split(',',$self->param('ht') || "");
     my $highlight_map = $self->get_highlight_map($cdb,$tree->tree);
     #my @compara_highlights; # not to be confused with COMPARA_HIGHLIGHTS list
+
+    my @ontology_terms = ();
     foreach my $ot_map (@$highlight_map){
       my $xref = $ot_map->{'xref'};
       if ( grep /^$xref$/, @highlight_tags ){
-        push (@highlights, 
+        push (@ontology_terms, 
           sprintf("%s,%s,%s", $xref, $ot_map->{'colour'}, join(',', @{$ot_map->{'members'}})) 
         );
       }
     }
+    push(@highlights, \@ontology_terms);
```

Comment:
This is pushing a reference to an array into the highlights array, where previously the code pushed in a string (produced by `sprintf` on line 287). Is this intentional?

Reply:
Yes, the previous code pushed a string into the `highlights` array. Notice, however, that it was doing so in a `foreach` loop; so if there ever were more than one such sting, they will be pushed into the `highlights` array one after the other. The code consuming the `highlights` array is not accounting for this; so any such hypothetical extra strings will be silently ignored.

It is probably because of that that Anne decided to [change the consuming code](https://github.com/EnsemblGenomes/eg-web-common/commit/de70161e4e3728b7793c48b8d5a200b2d001fbe4); but for some reason, neglected to change the code populating the highlights. I think her intention was that the `highlights` array will contain, as its ninth member, an array of strings. This also makes the length of the highlights array predictable, which I think is easier for consumption than just pushing an unknown number of strings directly into the `highlights` array.

Reply:
To add, Andrey's changes are migrating `highlights[8]` array element from string to array, because looking at the 3 code places this element is used, the migration has been started but not finished (so in a broken state):
* This line here fixes the element population (pushing strings to `highlights[8]` instead of `[8],[9],....`)
* `genetree.pm` (see [another PR](https://github.com/Ensembl/ensembl-webcode/pull/972/files)) was expecting string in `highlights[8]` (and was splitting it to array), after the changes it can use the array directly
* `genetree_legend.pm` (in this PR) already expected an array so no changes needed